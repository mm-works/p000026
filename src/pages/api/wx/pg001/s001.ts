import nextConnect from 'next-connect';
import { NextApiRequest, NextApiResponse, PageConfig } from 'next';
import anylogger from 'anylogger';
import '@mmstudio/an000042';
import an49 from '@mmstudio/an000049';
import a002 from '../../../../atoms/a002';

const logger = anylogger('pages/api/wx/s001');

export type Result = ITbswiper[];

/**
 * 获取首页轮播图列表
 */
const handler = a002<Result>();

handler.post(async (req, res) => {
	const db = an49();
	const dt = db<ITbswiper>('swiper');
	const data = await dt.select('*');
	res.status(200).json(data);
});

export const config = {} as PageConfig;

export default handler;
