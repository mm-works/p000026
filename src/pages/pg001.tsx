import { GetStaticProps, NextPage } from 'next';
import useSWR, { mutate } from 'swr';
import anylogger from 'anylogger';
import Head from 'next/head';

const logger = anylogger('pg001');

interface IProps {
	name: string
}

const pg001: NextPage<IProps> = (context) => {
	console.debug(context);	// anylogger would not print anything if configured with log4js, use console would be fine.
	const { data } = useSWR<{ time: string }>('api/s001');
	// const { data } = useSWR<{ time: string; }>('api/s001', { refreshInterval: 3000 });
	async function update() {
		await mutate('api/s001');
	}
	return (
		<>
			<Head>
				<title>mmstudio</title>
			</Head>
			<div>Hello {context.name} !</div>
			<br />
			<input type="button" value="update" onClick={update} />
			<br />
			<div>系统已启动: {data?.time}</div>
		</>
	);
};

export const getStaticProps: GetStaticProps<IProps> = async (content) => {
	logger.info(content);
	return Promise.resolve({
		props: {
			name: 'mmstudio',
		},
	});
};

export default pg001;
