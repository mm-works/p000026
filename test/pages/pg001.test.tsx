import React from 'react'
import { fireEvent, render } from '../testUtils'
import pg001 from '../../src/pages/pg001'

const PG001 = pg001 as () => JSX.Element
describe('Home page', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<PG001 />, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('clicking button triggers alert', () => {
    const { getByText } = render(<PG001 />, {})
    window.alert = jest.fn()
    fireEvent.click(getByText('update'))
  })
})
