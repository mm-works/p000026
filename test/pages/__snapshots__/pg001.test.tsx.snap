// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div>
    Hello  !
  </div>
  <br />
  <input
    type="button"
    value="update"
  />
  <br />
  <div>
    系统已启动: 
  </div>
</DocumentFragment>
`;
